<?xml version="1.0"?>
<launch>
    <!-- Lane Detection Parameters -->
    <arg name="lane_width" default="3.0"/>
    <arg name="max_detection_distance" default="10.0"/>
    <arg name="min_detection_distance" default="1.0"/>
    <arg name="enable_debug" default="true"/>
    
    <!-- HSV Color Parameters for White Lane Markings -->
    <arg name="white_lower_h" default="0"/>
    <arg name="white_lower_s" default="0"/>
    <arg name="white_lower_v" default="200"/>
    <arg name="white_upper_h" default="180"/>
    <arg name="white_upper_s" default="30"/>
    <arg name="white_upper_v" default="255"/>
    
    <!-- HSV Color Parameters for Yellow Lane Markings -->
    <arg name="yellow_lower_h" default="15"/>
    <arg name="yellow_lower_s" default="100"/>
    <arg name="yellow_lower_v" default="100"/>
    <arg name="yellow_upper_h" default="35"/>
    <arg name="yellow_upper_s" default="255"/>
    <arg name="yellow_upper_v" default="255"/>

    <!-- Lane Detection Node -->
    <node pkg="ashwini" type="lane_detection.py" name="lane_detection_node" output="screen">
        <!-- Basic Parameters -->
        <param name="lane_width" value="$(arg lane_width)"/>
        <param name="max_distance" value="$(arg max_detection_distance)"/>
        <param name="min_distance" value="$(arg min_detection_distance)"/>
        
        <!-- HSV Parameters -->
        <param name="white_lower_h" value="$(arg white_lower_h)"/>
        <param name="white_lower_s" value="$(arg white_lower_s)"/>
        <param name="white_lower_v" value="$(arg white_lower_v)"/>
        <param name="white_upper_h" value="$(arg white_upper_h)"/>
        <param name="white_upper_s" value="$(arg white_upper_s)"/>
        <param name="white_upper_v" value="$(arg white_upper_v)"/>
        <param name="yellow_lower_h" value="$(arg yellow_lower_h)"/>
        <param name="yellow_lower_s" value="$(arg yellow_lower_s)"/>
        <param name="yellow_lower_v" value="$(arg yellow_lower_v)"/>
        <param name="yellow_upper_h" value="$(arg yellow_upper_h)"/>
        <param name="yellow_upper_s" value="$(arg yellow_upper_s)"/>
        <param name="yellow_upper_v" value="$(arg yellow_upper_v)"/>
        
        <!-- Camera Topic Remapping -->
        <remap from="/camera/color/image_raw" to="/camera/color/image_raw"/>
        <remap from="/camera/aligned_depth_to_color/image_raw" to="/camera/aligned_depth_to_color/image_raw"/>
        <remap from="/camera/color/camera_info" to="/camera/color/camera_info"/>
    </node>

    <!-- Lane Costmap Plugin -->
    <node pkg="ashwini" type="lane_costmap_plugin.py" name="lane_costmap_plugin" output="screen">
        <param name="inflation_radius" value="0.5"/>
        <param name="cost_scaling_factor" value="10.0"/>
        <param name="update_frequency" value="5.0"/>
    </node>

</launch> 